File: scr/HelloWord.ts
console.log("Hello, World!");

/* 
    npm install -g typescript
    tsc HelloWord.ts
    node HelloWord.js
*/


// Simple demo
function greet(name: string, age: number): void {
    console.log(`Hello, ${name}! You are ${age} years old.`);
}

const userName: string = "Alice";
const userAge: number = 30;

greet(userName, userAge);

File: scr/HelloWord.rs
fn main() {
    // Print "Hello, World!" to the console
    println!("Hello, World!");
}

/*
    rustc HelloWord.rs
    ./HelloWord
*/


// Simple demo
fn main() {
    let name = String::from("Rust");
    
    // Pass reference to avoid taking ownership
    greet(&name); 

    // `name` is still valid here because we passed a reference
    println!("Again: {}", name);
}

fn greet(name: &str) {
    println!("Hello, {}!", name);
}